### PR TITLE
Extend stream excluder to be able to fail on wrong stream/version series

### PR DESCRIPTION
**Pull request by mfalesni**
**SHOULD FAIL ON BOTH DOWNSTREAMS TO SHOW IT WORKS**

Example usage:
```bash
py.test ... --check-stream downstream-52z
```

The version check is dependent on the stream check.

Should help us if an image swap that happened recently happens.

{{pytest: -v -m smoke --check-stream upstream }}

### CODE REVIEW

File: markers/stream_excluder.py [hunk 1]
```diff
@@ -21,6 +23,20 @@ def get_streams_id():
         return {"upstream"}
 
 
+def pytest_addoption(parser):
+    group = parser.getgroup('Specific stream smoke testing')
+    group.addoption('--check-stream',
+                    action='store',
+                    default=None,
+                    dest='check_stream',
+                    help='Specify "upstream" or "downstream" if you want to use this feature')
```
Comment (psav) [inline markers/stream_excluder.py]: Would it be possible to do "downstream-54z" and not just "downstream"?

**Comment by dajoRH** [comment]
Lint report for commit 162dd30e13c615176cba62b112e1994f4fb509b6:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: markers/stream_excluder.py (hunk 1 above)
Comment (mfalesni) [inline markers/stream_excluder.py]: py.test does not know about our shorthands. Only if I applied a regexp on it

**Comment by dajoRH** [comment]
Lint report for commit ce8bcfc158489cc0aaa0e3052f96b7b250977978:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
Seems working properly.

File: markers/stream_excluder.py [hunk 2]
```diff
@@ -71,3 +91,25 @@ def pytest_collection_modifyitems(session, config, items):
     # Just to print out the appliance's streams
     if not ssh_connection_failed:
         reporter(config).write("\nAppliance's streams: [{}]\n".format(", ".join(get_streams_id())))
+        # Bail out if the appliance stream or version do not match
+        check_stream = config.getvalue("check_stream")
```
Comment (seandst) [inline markers/stream_excluder.py]: I think you should default check_stream to `''`, declare its type as `str`, and then `check_stream = config.getvalue("check_stream").lower()` here.

File: markers/stream_excluder.py [hunk 3]
```diff
@@ -71,3 +91,25 @@ def pytest_collection_modifyitems(session, config, items):
...
+        if check_stream is not None:
+            match = STREAM_RE.match(check_stream)
+            if match is not None:
+                check_stream, major, minor = match.groups()
+                if major and minor:
+                    check_version_series = "{}.{}".join((major, minor))
+            if check_stream.lower() == "upstream":
+                if appliance_is_downstream():
+                    raise Exception("Appliance is downstream when we want upstream one!")
+            elif check_stream.lower() == "downstream":
```
Comment (seandst) [inline markers/stream_excluder.py]: Do we anything in `utils.version` that can help us out here? I'm wondering if the stream matching functionality might be generic enough to go in there.

Comment (seandst) [inline markers/stream_excluder.py]: And just to clarify, this is a generic question. Any time we compare against a string literal, I will wonder if there's value in having a python type to centralize that comparison behavior.

Comment (mfalesni) [inline markers/stream_excluder.py]: Hmm, I got an idea ...

**Comment by dajoRH** [comment]
Lint report for commit d4fbb92bc2ae8b63089a6cd0c20b951d5416c7c0:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by seandst** [comment]
Well that's way simpler now. :D

**Pull request merged**